packages/terra-navigation-layout/src/RoutingStack.jsx

```diff
@@ -107,11 +119,16 @@ class RoutingStack extends React.Component {
           path={routeData.path}
           key={routeData.path}
           render={() => (
-            <ComponentClass
-              {...ancestorProps}
-              {...routeData.componentProps}
-              routingStackDelegate={RoutingStackDelegate.create(delegateData)}
-              app={app}
+            <Route
```

Comment:
This additional wrapper is reseting the router context for the ComponentClass and anything rendered within it. 

Without the wrapper, anything that reads from the router context directly would only have access to the RoutingStack's location and lose access completely to the actual router's location. The actual router's location can be useful for matching NavLinks or any number of things, so we want to ensure those components get access to it. So we inject this path-less Route around the content with the location set to the router location to reset the context for the stuff inside it.

As I was writing that, I think I changed my mind regarding returning the current stackLocation inside the RoutingStackDelegate. I had changed it in this PR because I didn't think anyone would care about the RoutingStack's internal location, but I will continue to return it just to cover our bases.

Reply:
very tricky!